server/eclipse-project/src/main/java/us/freeandfair/corla/asm/AsmTransitions.java

```diff
@@ -0,0 +1,223 @@
+      new Pair<AsmState, AsmEvent>(AsmState.CountyDashboardState.AUDIT_BOARD_ESTABLISHED_STATE, 
+               AsmEvent.CountyDashboardEvent.COUNTY_UPLOAD_VERIFIED_BALLOT_MANIFEST_EVENT),
+        AsmState.CountyDashboardState.COUNTY_UPLOAD_VERIFIED_BALLOT_MANIFEST_TOO_LATE),
+    MANIFEST_RESTART_FROM_HASH_WRONG(
+       new Pair<AsmState, AsmEvent>(AsmState.CountyDashboardState.COUNTY_UPLOAD_VERIFIED_BALLOT_MANIFEST_HASH_WRONG, 
+                AsmEvent.CountyDashboardEvent.SKIP_EVENT),
+         AsmState.CountyDashboardState.AUDIT_BOARD_ESTABLISHED_STATE),
+    MANIFEST_RESTART_FROM_TYPE_WRONG(
+       new Pair<AsmState, AsmEvent>(AsmState.CountyDashboardState.COUNTY_UPLOAD_VERIFIED_BALLOT_MANIFEST_FILE_TYPE_WRONG, 
+                AsmEvent.CountyDashboardEvent.SKIP_EVENT),
+         AsmState.CountyDashboardState.AUDIT_BOARD_ESTABLISHED_STATE),
+    MANIFEST_RESTART_FROM_INTERRUPTED(
+       new Pair<AsmState, AsmEvent>(AsmState.CountyDashboardState.COUNTY_UPLOAD_VERIFIED_BALLOT_MANIFEST_DATA_TRANSMISSION_INTERRUPTED, 
+                AsmEvent.CountyDashboardEvent.SKIP_EVENT),
+         AsmState.CountyDashboardState.AUDIT_BOARD_ESTABLISHED_STATE);
+    // @todo kiniry Add all remaining transitions.
+    
+    /**
+     * The pair holding a single transition.
+     */
+    protected final Pair<Pair<AsmState, AsmEvent>, AsmState> my_pair;
+    
+    /**
+     * Create a transition.
+     * @param the_pair the (current state, event) pair.
+     * @param the_state the state transitioned to when the pair is witnessed.
+     */
+    CountyDashboardTransitions(final Pair<AsmState, AsmEvent> the_pair,
+                               final AsmState the_state) {
+      my_pair = new Pair<Pair<AsmState, AsmEvent>, AsmState>(the_pair, the_state);
+    }
+  }
+  
+  /**
+   * The Audit Board Dashboard's transition function.
+   * @trace asm.audit_board_dashboard_next_state
+   */
+  enum AuditBoardDashboardTransitions implements AsmTransitions {
+    START(
+        new Pair<AsmState, AsmEvent>(AsmState.AuditBoardDashboardState.INITIAL_STATE, 
+        AsmEvent.AuditBoardDashboardEvent.SKIP_EVENT),
+        AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE),
+    REPORT_MARKING(
+        new Pair<AsmState, AsmEvent>(AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE, 
+        AsmEvent.AuditBoardDashboardEvent.REPORT_MARKINGS_EVENT),
+        AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE),
+    REPORT_BNF(
+        new Pair<AsmState, AsmEvent>(AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE, 
+        AsmEvent.AuditBoardDashboardEvent.REPORT_BALLOT_NOT_FOUND_EVENT),
+        AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE),
+    SUBMIT_AIR(
+        new Pair<AsmState, AsmEvent>(AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE, 
+        AsmEvent.AuditBoardDashboardEvent.SUBMIT_AUDIT_INVESTIGATION_REPORT_EVENT),
+        AsmState.AuditBoardDashboardState.AUDIT_IN_PROGRESS_STATE),
```

Comment:
For example, the CORLA2 document "Discrepancy Investigation procedure & Tracking metrics" notes
one procedure that seems to address a number of possible attacks. E.g. if the audit board wants to “cure” a discrepancy that was a result of retrieving the wrong ballot, they can find both the ballot corresponding to the original sample, and a CVR corresponding to the ballot they say they pulled by mistake. If both match, I think it may be reasonable to eliminate the original discrepancy. This would require the ability for the audit board to both re-enter a new ACVR for the original selection, and for them to enter the BallotID of the ballot they seem to have actually found in the first place, and enter an ACVR for it. ACVRs for both ballots should match the corresponding CVRs.